### ISSUE TITLE
some of your val by lazy should probably be a fun assertSomething instead

### ISSUE THREAD

**Issue by RedJocker** [hatepoint/phrases-hyperskill-tests#4]
doing assertions on a lazy initialization block makes sense when you have something, usually a view, that has to be initialized with some expected property value and you want to make sure when you use that widget that it is initialized with that value.

So this makes sense:

```

protected val reminderTv: TextView by lazy {
        val view = activity.findViewByString<TextView>("reminderTextView")
        val messageInitialText = "The reminderTextView has a wrong initial text"
        val expectedInitialText = "No reminder set"
        val actualInitialText = view.text.toString()
        assertEquals(messageInitialText, expectedInitialText, actualInitialText)

        view
    }

```

but this it is a very different case when you are writing something focusing in the assertions and not on using the property like this:

```
protected val recyclerViewClick: RecyclerView by lazy {
        val view: RecyclerView = activity.findViewByString("recyclerView")
        val actualAmount = view.adapter!!.itemCount
        val expectedAmount = actualAmount - 1
        view.findViewByString<TextView>("deleteTextView").clickAndRun()
        assertEquals("The recyclerView didn't remove item after clicking 'Delete'.", expectedAmount, view.adapter!!.itemCount)
        view
    }
```

This should better be either a @test fun on StageUnitTest or a fun on PhrasesUnitTest depending on how recurrent this case is. 
If it is a case for PhrasesUnitTest then you can also think about making an extension function like fun RecyclerView.assertDeleteItem() {...}

In short: use val by lazy when you intend to use that property somewhere else. Do assertions on the init block when that property has to be initialized with some values. If you just want to make assertions use a fun.

**Comment by hatepoint** [comment]
I fixed these val's but I might need you to take a look whether they're correct. I'll reply with a link to commit when done

**Comment by hatepoint** [comment]
43f87bf56845e48bd06f1c8538bfa6ad57c6929b is it okay now?

**Comment by RedJocker** [comment]
yes, at a first glance it looks better. 
I see you put the solution too. 
I'll take a closer look later on, but this issue seems to be closed.
